src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1137,6 +1143,11 @@ private void AppendNewBlock(BasicBlock block, bool linkToPrevious = true)
                 }
             }
 
+            if (block.Ordinal != -1)
+            {
+                throw ExceptionUtilities.Unreachable;
```

Comment:
Why is this not just an assert?

Reply:
> Why is this not just an assert?

Because we are about to add the same block twice into the graph in different positions - corruption of the graph.

---
In reply to: [188748402](https://github.com/dotnet/roslyn/pull/26903#discussion_r188748402) [](ancestors = 188748402)